tools/skbstat.py

```diff
@@ -0,0 +1,996 @@
+#!/usr/bin/python
+# @lint-avoid-python-3-compatibility-imports
+#
+# skbstat Trace skb events(ingress/egress/consume/drop) of kernel.
+#         For Linux, uses BCC, eBPF. Embedded C.
+#
+# USAGE: skbstat [-h] [-i INTERVAL] [-s SADDR] [-d DADDR] [-N ADDR]
+#                  [-S SPORT] [-D DPORT] [-P PORT] [-p PROTO] [--input]
+#                                    [--output] [--consume]
+#
+# Copyright (c) 2021 Dongdong Wang <wangdongdong.6@bytedance.com>
+# Licensed under the Apache License, Version 2.0 (the "License")
+#
+# 10-MAR-2021  Dongdong Wang Created this.
+
+from __future__ import print_function
+from bcc import BPF
+import argparse
+import ipaddress
+from time import strftime
+from struct import unpack
+from time import sleep
+
+# arguments
+examples = """examples:
+    ./skbstat                               # dump all skb dropping backtraces
```

Comment:
Ok, I'll fix that error, thx.

Reply:
`skbstat` is a little bit different with `tcpdrop`. It probe `kfree_skb` function call which leads to packet drops. The packet can be any protocol (TCP/UDP/ICMP...), not just TCP.
There are so many `kfree_skb` routines in kernel, so it may not be easy to tell the user why the packet got lost.  :-(
`skbstat` will provide filter arguments (source/dest IP ...) to let the users see the drop backtrace of the packet which  they really care about.